lib/ansible/parsing/yaml/objects.py

```diff
@@ -53,7 +54,17 @@ def _set_ansible_position(self, obj):
     ansible_pos = property(_get_ansible_position, _set_ansible_position)
 
 
-class AnsibleMapping(AnsibleBaseYAMLObject, dict):
+# try to always use orderddict with yaml, after py3.6 the dict type already does this
+odict = dict
+if sys.version_info[:2] < (3, 7):
```

Comment:
I just realized this can be simplified, although I can also see why you might want to keep it as-is:

```suggestion
if sys.version_info < (3, 7):
```